.github/workflows/docker_image.yml

```diff
@@ -51,48 +59,133 @@ jobs:
       - name: Check out repository
         uses: actions/checkout@v4
 
+      - name: Prepare
+        run: |
+          platform=${{ matrix.platform }}
+          echo "PLATFORM_PAIR=${platform//\//-}" >> $GITHUB_ENV
+
       - name: Setup Docker buildx
-        uses: docker/setup-buildx-action@v3.0.0
+        uses: docker/setup-buildx-action@v3.10.0
 
+      # Extract metadata (tags, labels, annotations) for Docker
+      # https://github.com/docker/metadata-action
+      - name: Extract Docker metadata
+        id: meta
+        uses: docker/metadata-action@v5.7.0
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          labels: |
+            org.opencontainers.image.title=Luanti
+            org.opencontainers.image.vendor=Luanti
+            org.opencontainers.image.description=Luanti (formerly Minetest) is an open source voxel game-creation platform with easy modding and game creation
+            org.opencontainers.image.licenses=LGPL-2.1-only
+          annotations: |
+            org.opencontainers.image.title=Luanti
+            org.opencontainers.image.vendor=Luanti
+            org.opencontainers.image.description=Luanti (formerly Minetest) is an open source voxel game-creation platform with easy modding and game creation
+            org.opencontainers.image.licenses=LGPL-2.1-only
+      
       # Login against the Docker registry except on PR
       # https://github.com/docker/login-action
       - name: Log into registry ${{ env.REGISTRY }}
         if: github.event_name != 'pull_request'
-        uses: docker/login-action@v3.0.0
+        uses: docker/login-action@v3.4.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      
+      - name: Get image name
+        run:
+          echo "IMAGE_NAME_LOWERCASE"="${IMAGE_NAME,,}" >> $GITHUB_ENV
+
+      - name: Build and push by digest
+        id: build
+        uses: docker/build-push-action@v6.15.0
+        with:
+          context: .
+          platforms: ${{ matrix.platform }}
+          labels: ${{ steps.meta.outputs.labels }}
+          annotations: ${{ steps.meta.outputs.annotations }}
+          tags: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME_LOWERCASE }}
+          outputs: type=image,push-by-digest=true,name-canonical=true,push=${{ github.event_name != 'pull_request' }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          no-cache: ${{ (github.event_name == 'workflow_dispatch' && !inputs.use_cache) || startsWith(github.ref, 'refs/tags/') }}
```

Comment:
what does this line do

Reply:
The `no-cache` line? Prevent the Docker build cache from being used on tag builds and when the workflow is dispatched without the "use_cache" option.

Reply:
If let's say `alpine:3.19` effectively points to `3.19.8` at the time of a Luanti release, have a massive security vulnerability that is discovered and patched in `3.19.9` you can trigger the workflow on the Luanti tag so the image is rebuilt with the most recent `3.19` patch release.